streams/src/main/java/org/apache/kafka/streams/processor/internals/namedtopology/TopologyConfig.java

```diff
@@ -0,0 +1,176 @@
+        this.eosEnabled = StreamThread.eosEnabled(globalAppConfigs);
+
+        if (isTopologyOverride(MAX_TASK_IDLE_MS_CONFIG, topologyOverrides)) {
+            maxTaskIdleMs = getLong(MAX_TASK_IDLE_MS_CONFIG);
+            log.info("Topology {} is overriding {} to {}", topologyName, MAX_TASK_IDLE_MS_CONFIG, maxTaskIdleMs);
+        } else {
+            maxTaskIdleMs = globalAppConfigs.getLong(MAX_TASK_IDLE_MS_CONFIG);
+        }
+
+        if (isTopologyOverride(TASK_TIMEOUT_MS_CONFIG, topologyOverrides)) {
+            taskTimeoutMs = getLong(TASK_TIMEOUT_MS_CONFIG);
+            log.info("Topology {} is overriding {} to {}", topologyName, TASK_TIMEOUT_MS_CONFIG, taskTimeoutMs);
+        } else {
+            taskTimeoutMs = globalAppConfigs.getLong(TASK_TIMEOUT_MS_CONFIG);
+        }
+
+        if (isTopologyOverride(BUFFERED_RECORDS_PER_PARTITION_CONFIG, topologyOverrides)) {
+            maxBufferedSize = getInt(BUFFERED_RECORDS_PER_PARTITION_CONFIG);
+            log.info("Topology {} is overriding {} to {}", topologyName, BUFFERED_RECORDS_PER_PARTITION_CONFIG, maxBufferedSize);
+        } else {
+            maxBufferedSize = globalAppConfigs.getInt(BUFFERED_RECORDS_PER_PARTITION_CONFIG);
+        }
+
+        if (isTopologyOverride(DEFAULT_TIMESTAMP_EXTRACTOR_CLASS_CONFIG, topologyOverrides)) {
+            timestampExtractorSupplier = () -> getConfiguredInstance(DEFAULT_TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
+            log.info("Topology {} is overriding {} to {}", topologyName, DEFAULT_TIMESTAMP_EXTRACTOR_CLASS_CONFIG, getClass(DEFAULT_TIMESTAMP_EXTRACTOR_CLASS_CONFIG));
+        } else {
+            timestampExtractorSupplier = () -> globalAppConfigs.getConfiguredInstance(DEFAULT_TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
+        }
+
+        if (isTopologyOverride(DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG, topologyOverrides)) {
+            deserializationExceptionHandlerSupplier = () -> getConfiguredInstance(DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG, DeserializationExceptionHandler.class);
+            log.info("Topology {} is overriding {} to {}", topologyName, DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG, getClass(DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG));
+        } else {
+            deserializationExceptionHandlerSupplier = () -> globalAppConfigs.getConfiguredInstance(DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG, DeserializationExceptionHandler.class);
+
+        }
+    }
+
+    /**
+     * @return true if there is an override for this config in the properties of this NamedTopology. Applications that
+     *         don't use named topologies will just refer to the global defaults regardless of the topology properties
+     */
+    private boolean isTopologyOverride(final String config, final Properties topologyOverrides) {
+        return topologyName != null && topologyOverrides.containsKey(config);
```

Comment:
Should we have warnning log for users to let them know their properties doesn't get overridden because of no topology name set? Otherwise, the user will never know it, right?

Reply:
Yeah, this NamedTopology stuff is still an experimental feature and hasn't gone through a KIP so we shouldn't expect the topology name to be set anyways (yet)

Reply:
Added a TODO for now